scripts/ice-road-pipeline.py

```diff
@@ -124,15 +164,170 @@
+    if shp_fp_rfl:
+
+        # Make sure R is installed
+        proc = Popen(["which", "R"],stdout=PIPE,stderr=PIPE)
+        exit_code = proc.wait()
+        if exit_code != 0:
+            raise Exception("Please install R on this system and packages: raster, lidR, rlas, dplyr, readr, terra,sf, data.table.")
+
+        # Check whether dem directory exists, if not, make one
+        ssa_dir = f'{results_dir}/ssa-calc'
+        if not os.path.exists(ssa_dir):
+            os.makedirs(ssa_dir)
+
+        # Displaying the parent directory of the script
+        scripts_dir = os.path.dirname(__file__)
+
+        # Load in the ASP translational-only matrix from pc-align tool
+        asp_matrix_fp = glob(f'{ice_dir}/results/pc-align-translation-only/temp-log*.txt')[0]
+        with open(asp_matrix_fp, 'r') as f:
+            lines = f.readlines()
+            for line in lines:
+                if "Translation vector (North-East-Down, meters): Vector3" in line: 
+                    n_e_d_shift = re.findall("[+-]?\d+\.\d+", line)
+                    n_e_d_shift = ','.join(n_e_d_shift)
+                    break
+
+        # get crs
+        ref_raster = rasterio.open(snow_tif)
+        crs = ref_raster.crs
+        pix_size, _ = ref_raster.res
+        crs = str(crs).split(":",1)[1]    
+        ras_meta = ref_raster.profile
+
+        # Compute slope and aspect from snow-on lidar
+        slope_fp = join(ice_dir, f'{basename(in_dir)}-snowon_slope.tif')
+        aspect_fp = join(ice_dir, f'{basename(in_dir)}-snowon_aspect.tif')
+        os.system(f'gdaldem slope -compute_edges {snow_tif} {slope_fp} -q')
+        os.system(f'gdaldem aspect -compute_edges -zero_for_flat {snow_tif} {aspect_fp} -q')
+        slope = rio.open_rasterio(slope_fp, masked=True)
+        aspect = rio.open_rasterio(aspect_fp, masked=True)
+
+        # Compute surface normal vector from snow-on lidar
+        ni_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_i.tif')
+        nj_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_j.tif')
+        nk_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_k.tif')
+        n_i = np.sin(np.radians(aspect)) * np.sin(np.radians(slope))
+        n_j = np.cos(np.radians(aspect)) * np.sin(np.radians(slope)) 
+        n_k = np.cos(np.radians(slope))
+        n_i.rio.to_raster(ni_fp)
+        n_j.rio.to_raster(nj_fp)
+        n_k.rio.to_raster(nk_fp)        
+        
+        # Calcs calibration stats from target
+        output_csv = f'{ssa_dir}/all-calibration-rfl.csv'
+        subprocess.call(["Rscript", 
```

Comment:
Can we add a logging step to directly log what command is being done so we can trouble shoot?